test/livebook_web/plugs/proxy_plug_test.exs

```diff
@@ -0,0 +1,109 @@
+defmodule LivebookWeb.ProxyPlugTest do
+  use LivebookWeb.ConnCase, async: true
+
+  require Phoenix.LiveViewTest
+  import Livebook.SessionHelpers
+  import Livebook.AppHelpers
+
+  alias Livebook.{Notebook, Runtime, Session, Sessions}
+
+  describe "session" do
+    test "returns error when session doesn't exist", %{conn: conn} do
+      session_id = Livebook.Utils.random_long_id()
+
+      assert_error_sent 404, fn ->
+        get(conn, "/sessions/#{session_id}/proxy/foo/bar")
+      end
+    end
+
+    test "returns error when runtime is disconnected", %{conn: conn} do
+      {:ok, session} = Sessions.create_session()
+
+      assert_error_sent 404, fn ->
+        get(conn, "/sessions/#{session.id}/proxy/foo/bar")
+      end
+    end
+
+    test "returns the proxied response defined in notebook", %{conn: conn} do
+      notebook = %{Notebook.new() | name: "My Notebook"}
+
+      {:ok, session} = Sessions.create_session(notebook: notebook)
+      {:ok, runtime} = Runtime.Embedded.new() |> Runtime.connect()
+
+      Session.set_runtime(session.pid, runtime)
+      Session.subscribe(session.id)
+
+      cell_id =
+        insert_text_cell(
+          session.pid,
+          insert_section(session.pid),
+          :code,
+          """
+          Kino.Proxy.listen(fn conn ->
+            conn
+            |> Plug.Conn.put_resp_header("content-type", "application/text;charset=utf-8")
+            |> Plug.Conn.send_resp(200, "used " <> conn.method <> " method")
+          end)
+          """
+        )
```

Comment:
Let's build the notebook by hand here too, and pass to `Sessions.create_session(notebook: notebook)` :)